main.tf

```diff
@@ -38,7 +44,7 @@ resource "aws_iam_role_policy_attachment" "lambda" {
 # CloudWatch Log group
 #---------------------------------------------------------------------------------------------------
 resource "aws_cloudwatch_log_group" "this" {
-  name              = "/aws/lambda/${aws_lambda_function.this.function_name}"
+  name              = "/aws/lambda/${var.function_name}"
```

Comment:
Why are we switching from souring the function name from the function to the variable here?

Reply:
To avoid a cyclic reference